qiskit-superstaq/qiskit_superstaq/superstaq_job.py

```diff
@@ -54,10 +54,55 @@ def _wait_for_results(self, timeout: float, wait: float = 5) -> List[Dict[str, D
         Returns:
             Results from the job.
         """
-
         self.wait_for_final_state(timeout, wait)  # should call self.status()
         return [self._job_info[job_id] for job_id in self._job_id.split(",")]
 
+    def _arrange_counts(
+        self, counts: Dict[str, int], circ_meas_bit_indices: List[int], num_clbits: int
+    ) -> Dict[str, int]:
+        """Arranges the classical bit strings from job counts to match classical register.
+
+        Args:
+            counts: The raw counts from a job result.
+            circ_meas_bit_indices: The indices of the measured qubits.
+            num_clbits: The number of classical bits for the corresponding job circuit.
+
+        Returns:
+            A dictionary with the updated counts keys.
+        """
+        arranged_counts = {}
+        for key in counts.keys():
+            updated_key = "0" * num_clbits
+            for counter, index in enumerate(circ_meas_bit_indices):
+                updated_key = updated_key[:index] + key[counter] + updated_key[index + 1 :]
+            arranged_counts[updated_key] = counts[key]
+
+        return arranged_counts
+
+    def _get_clbit_indices(self, index: int) -> List[int]:
+        """Helper method to update the measurement indices from the compiled circuit.
+
+        Args:
+            index: The index of the compiled circuit to get indices from.
+
+        Returns:
+            The specific measurement indices of the circuit with label `index` in
+            the job.
+        """
+        input_circuit = self._get_circuits("input_circuit", index=index)
```

Comment:
could this just be
```suggestion
        input_circuit = self.input_circuits(index)
```
?